tests/unit/amo/components/TestAddon.js

```diff
@@ -761,23 +763,26 @@ describe(__filename, () => {
     expect(button.prop('disabled')).toEqual(false);
   });
 
-  it('disables install switch for unsupported clients', () => {
+  it('disables install btn for incompatibility with firefox version', () => {
     const root = shallowRender({
-      getClientCompatibility: getClientCompatibilityFalse,
+      getClientCompatibility: () => ({
+        compatible: false,
+        downloadUrl: 'https://www.seamonkey-project.org',
+        reason: INCOMPATIBLE_UNDER_MIN_VERSION,
+      }),
```

Comment:
I would not change this test like this. I would keep `getClientCompatibilityFalse` and update the assertions:

```diff
diff --git a/tests/unit/amo/components/TestAddon.js b/tests/unit/amo/components/TestAddon.js
index cc9cbd1..567aad0 100644
--- a/tests/unit/amo/components/TestAddon.js
+++ b/tests/unit/amo/components/TestAddon.js
@@ -763,15 +763,15 @@ describe(__filename, () => {
     expect(button.prop('disabled')).toEqual(false);
   });
 
-  it('disables install btn for incompatibility with firefox version', () => {
+  it('displays a button to get firefox for unsupported clients', () => {
     const root = shallowRender({
-      getClientCompatibility: () => ({
-        compatible: false,
-        downloadUrl: 'https://www.seamonkey-project.org',
-        reason: INCOMPATIBLE_UNDER_MIN_VERSION,
-      }),
+      getClientCompatibility: getClientCompatibilityFalse,
     });
-    expect(root.find(InstallButton).prop('disabled')).toBe(true);
+    expect(root.find(Button)).toHaveLength(1);
+    expect(root.find(Button).prop('children'))
+      .toEqual('Only with Firefox - Get Firefox Now!');
+    expect(root.find(Button).prop('href'))
+      .toEqual('https://www.mozilla.org/firefox/new/?utm_source=addons.mozilla.org&utm_medium=referral&utm_campaign=non-fx-button&utm_content=install-addon-button');
   });
 
   it('hides banner on non firefox clients and displays firefox download btn', () => {

```

Reply:
ah well, we also need a test similar to yours:

```js
  it('disables install button when addon is incompatible with firefox version', () => {
    const root = shallowRender({
      getClientCompatibility: () => ({
        compatible: false,
        reason: INCOMPATIBLE_UNDER_MIN_VERSION,
      }),
    });
    expect(root.find(InstallButton).prop('disabled')).toBe(true);
  });
```

Both are needed. I think having a very small diff in a test case is easier to understand but feel free to do what works best for you.

Reply:
thanks @willdurand  - i think i can clean this up a little bit more